notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb

```diff
@@ -0,0 +1,472 @@
+      },
+      "outputs": [],
+      "source": [
+        "import bigframes\n",
+        "import bigframes.pandas\n",
+        "import pandas as pd\n",
+        "from google.cloud import bigquery\n",
+        "from vertexai.resources.preview.feature_store import (Feature, FeatureGroup,\n",
+        "                                                      offline_store)\n",
+        "from vertexai.resources.preview.feature_store import utils as fs_utils"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ac036ecfbc32"
+      },
+      "outputs": [],
+      "source": [
+        "BQ_DATASET_ID = \"fhfv_dataset_unique\"  # @param {type:\"string\"}\n",
+        "BQ_TABLE_ID = \"fhfv_table_unique\"  # @param {type:\"string\"}\n",
+        "BQ_TABLE_URI = f\"{PROJECT_ID}.{BQ_DATASET_ID}.{BQ_TABLE_ID}\"\n",
+        "\n",
+        "FEATURE_GROUP_ID = \"fhfv_fg_unique\"  # @param {type:\"string\"}\n",
+        "\n",
+        "DATA_SOURCE = \"gs://cloud-samples-data-us-central1/vertex-ai/feature-store/datasets/movie_prediction.csv\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "cd580a0679ce"
+      },
+      "source": [
+        "### Write BigQuery table for use in Feature Registry"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1ff4481243a8"
+      },
+      "outputs": [],
+      "source": [
+        "bigframes.options.bigquery.project = PROJECT_ID\n",
+        "bigframes.options.bigquery.location = LOCATION\n",
+        "df = bigframes.pandas.read_csv(DATA_SOURCE)\n",
+        "df[\"timestamp\"] = pd.to_datetime(df[\"timestamp\"], utc=True)\n",
+        "df = df.rename(columns={\"timestamp\": \"feature_timestamp\"})\n",
+        "df.to_gbq(BQ_TABLE_URI, if_exists=\"replace\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "818a36b9da86"
+      },
+      "source": [
+        "### Create Feature Registry Features"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1f915ddd4669"
+      },
+      "outputs": [],
+      "source": [
+        "fg: FeatureGroup = FeatureGroup.create(\n",
+        "    f\"{FEATURE_GROUP_ID}\",\n",
+        "    fs_utils.FeatureGroupBigQuerySource(\n",
+        "        uri=f\"bq://{BQ_TABLE_URI}\", entity_id_columns=[\"users\"]\n",
+        "    ),\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0dba1c02883c"
+      },
+      "outputs": [],
+      "source": [
+        "movie_feature: Feature = fg.create_feature(\"movies\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c71964219962"
+      },
+      "source": [
+        "### Fetch with same entity at multiple timestamps"
+      ]
+    },
+    {
```

Comment:
Add a description of the code before this block. For example, "Fetch feature values for the same entity at multiple timestamps."